packages/assets-controllers/src/MultichainAssetsController/MultichainAssetsController.test.ts

```diff
@@ -0,0 +1,1539 @@
+        parentCapability: 'snap_dialog',
+      },
+      snap_getBip32Entropy: {
+        caveats: [
+          {
+            type: 'permittedDerivationPaths',
+            value: [
+              {
+                curve: 'ed25519',
+                path: ['m', "44'", "501'"],
+              },
+            ],
+          },
+        ],
+        date: 1736869806348,
+        id: 'WiOkUgu4Y89p2youlm7ro',
+        invoker: 'local:http://localhost:8080',
+        parentCapability: 'snap_getBip32Entropy',
+      },
+      snap_getPreferences: {
+        caveats: null,
+        date: 1736869806349,
+        id: 'LmYGhkzdyDWmhAbM1UJfx',
+        invoker: 'local:http://localhost:8080',
+        parentCapability: 'snap_getPreferences',
+      },
+      snap_manageAccounts: {
+        caveats: null,
+        date: 1736869806349,
+        id: 'BaiVanA9U-BIfbCgYpeo6',
+        invoker: 'local:http://localhost:8080',
+        parentCapability: 'snap_manageAccounts',
+      },
+      snap_manageState: {
+        caveats: null,
+        date: 1736869806349,
+        id: 'oIJeVj2SsWh6uFam1RMi4',
+        invoker: 'local:http://localhost:8080',
+        parentCapability: 'snap_manageState',
+      },
+    };
+    const mockGetMetadataResponse: AssetMetadataResponse | undefined = {
+      assets: {
+        'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:501': {
+          name: 'Solana2',
+          symbol: 'SOL',
+          fungible: true,
+          iconUrl: 'url1',
+          units: [{ name: 'Solana2', symbol: 'SOL', decimals: 9 }],
+        },
+      },
+    };
+
+    mockSnapHandleRequest
+      .mockReturnValueOnce(mockAssetsResponse)
+      .mockReturnValueOnce(mockGetMetadataReturnValue)
+      .mockReturnValueOnce(mockGetMetadataResponse);
+
+    mockGetPermissions
+      .mockReturnValueOnce(mockSnapPermissionReturnVal)
+      .mockReturnValueOnce(mockGetPermissionsReturnValue[1])
+      .mockReturnValueOnce(mockGetPermissionsReturnValue[2])
+      .mockReturnValueOnce(mockGetPermissionsReturnValue[3])
+      .mockReturnValueOnce(mockGetPermissionsReturnValue[4])
+      .mockReturnValueOnce(mockGetPermissionsReturnValue[5])
+      .mockReturnValueOnce(mockGetPermissionsReturnValue[6]);
+
+    messenger.publish(
+      'AccountsController:accountAdded',
+      mockSolanaAccount as unknown as InternalAccount,
+    );
+
+    await advanceTime({ clock, duration: 1 });
+
+    expect(mockSnapHandleRequest).toHaveBeenCalledTimes(3);
+
+    expect(controller.state).toStrictEqual({
+      accountsAssets: {
+        [mockSolanaAccount.id]: mockAssetsResponse,
+      },
+      assetsMetadata: {
+        ...mockGetMetadataResponse.assets,
+        ...mockGetMetadataReturnValue.assets,
+      },
+    });
+  });
+
+  it('updates metadata in state successfully one call to fetch metadata fails', async () => {
+    const { controller, messenger, mockSnapHandleRequest, mockGetPermissions } =
+      setupController();
+
+    const mockAssetsResponse = [
```

Comment:
I would also rename it to match the actual handler call:

```suggestion
    const mockHandleRequestOnAssetsLookupResponse = [
```